### PR TITLE
Fix EZP-24303: Refactor ContentType form management

### PR DESCRIPTION
**Pull request by lolautruche**
> https://jira.ez.no/browse/EZP-24303
> https://jira.ez.no/browse/EZP-24306

* Added `FieldTypeFormMapperInterface`
* Implemented a registry for field type form mappers
* Added a compiler pass to register field type form mappers
* Added FieldType form mapper for `ezstring`

### CODE REVIEW

File: lib/FieldType/FieldTypeFormMapperInterface.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+namespace EzSystems\RepositoryForms\FieldType;
+
+use EzSystems\RepositoryForms\Data\FieldDefinitionData;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * Interface for FieldType form mappers.
+ *
+ * FieldType form mappers makes it possible to "map" and adapt forms related to FieldTypes (e.g. FieldDefinition forms).
```
Comment (bdunogier) [inline lib/FieldType/FieldTypeFormMapperInterface.php]: What about this ?

"It maps a FieldType's specifics to its editing Forms" (editing, or something else, more specific)

**Comment by lolautruche** [comment]
Review ping @andrerom @pspanja @yannickroger @dpobel

**Comment by bdunogier** [comment]
Nice, +1.

**Comment by lolautruche** [comment]
@dpobel take a look at b45b443 :wink:

File: tests/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+$file = __DIR__.'/../vendor/autoload.php';
+if (!file_exists($file)) {
+    throw new RuntimeException("Install dependencies using composer to run the test suite.");
+}
+
+$autoload = require_once $file;
```
Comment (andrerom) [inline tests/bootstrap.php]: nw

**Comment by lolautruche** [comment]
@dpobel you mean modifying the public API? Are you insane? :smile:

**Comment by lolautruche** [comment]
@dpobel Moved responsibility to translator. Translation file will go to PlatformUIBundle.

**Pull request merged**